opensrp-reveal/src/main/java/org/smartregister/reveal/presenter/ListTaskPresenter.java

```diff
@@ -638,4 +656,61 @@ public boolean isChangeMapPosition() {
     public void setChangeMapPosition(boolean changeMapPosition) {
         this.changeMapPosition = changeMapPosition;
     }
+
+    public void filterTasks(TaskFilterParams filterParams) {
+        filteredFeatureCollection = new ArrayList<>();
+        Set<String> filterStatus = filterParams.getCheckedFilters().get(Filter.STATUS);
+        Set<String> filterTaskCode = filterParams.getCheckedFilters().get(Filter.CODE);
+        Set<String> filterInterventionUnitTasks = getInterventionUnitCodes(filterParams.getCheckedFilters().get(Filter.INTERVENTION_UNIT));
+        Pattern pattern = Pattern.compile("~");
+        for (Feature feature : featureCollection.features()) {
+            boolean matches = true;
+            if (filterStatus != null) {
+                matches = feature.hasProperty(Properties.TASK_BUSINESS_STATUS) && filterStatus.contains(feature.getStringProperty(Properties.TASK_BUSINESS_STATUS));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'Properties.TASK_BUSINESS_STATUS' due to existing static import 'org.smartregister.reveal.util.Constants.Properties.TASK_BUSINESS_STATUS'](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=4684568)